api/src/main/java/io/grpc/TlsChannelCredentials.java

```diff
@@ -119,6 +242,126 @@ public Builder requireFakeFeature() {
       return this;
     }
 
+    /**
+     * Use the provided certificate chain and private key as the client's identity. Generally they
+     * should be PEM-encoded and the key is an unencrypted PKCS#8 key (file headers have "BEGIN
+     * CERTIFICATE" and "BEGIN PRIVATE KEY").
+     */
+    public Builder keyManager(File certChain, File privateKey) throws IOException {
+      return keyManager(certChain, privateKey, null);
+    }
+
+    /**
+     * Use the provided certificate chain and possibly-encrypted private key as the client's
+     * identity. Generally they should be PEM-encoded and the key is a PKCS#8 key. If the private
+     * key is unencrypted, then password must be {@code null}.
+     */
+    public Builder keyManager(File certChain, File privateKey, String privateKeyPassword)
+        throws IOException {
+      InputStream certChainIs = new FileInputStream(certChain);
+      try {
+        InputStream privateKeyIs = new FileInputStream(privateKey);
```

Comment:
I believe we can use "try-with-resources" here and similar places?

Reply:
I think try-with-resources is still off limits due to Android. (It may actually be safe now, but there's complications with suppressed exceptions and there's multiple Android toolchains and tracking each feature is hard. All this will change soon anyway once we drop Java 7 support.)